test/modules/home/home_interactor_test.dart

```diff
@@ -0,0 +1,163 @@
+    late MockSurveyRepository surveyRepository;
+    late MockAuthRepository authRepository;
+    late BehaviorSubjectGenerator generator;
+
+    beforeEach(() {
+      generator = BehaviorSubjectGenerator();
+
+      delegate = MockHomeInteractorDelegate();
+      when(delegate.authenticatedUserDidGet)
+          .thenAnswer((realInvocation) => generator.make(1));
+      when(delegate.surveysDidFetchFromCached)
+          .thenAnswer((realInvocation) => generator.make(2));
+      when(delegate.surveysDidFailToFetchFromCached)
+          .thenAnswer((realInvocation) => generator.make(3));
+      when(delegate.surveysDidFetchFromRemote)
+          .thenAnswer((realInvocation) => generator.make(4));
+      when(delegate.surveysDidFailToFetchFromRemote)
+          .thenAnswer((realInvocation) => generator.make(5));
+
+      surveyRepository = MockSurveyRepository();
+      locator.registerSingleton<SurveyRepository>(surveyRepository);
+
+      authRepository = MockAuthRepository();
+      locator.registerSingleton<AuthRepository>(authRepository);
+
+      interactor = HomeInteractorImpl();
+      interactor.delegate = delegate;
+    });
+
+    describe("its getAuthenticatedUser is called", () {
+      final user = UserInfo();
+      beforeEach(() {
+        when(authRepository.user).thenReturn(user);
+        interactor.getAuthenticatedUser();
+      });
+
+      it("triggers delegate's authenticatedUserDidGet to emit", () {
+        expect(delegate.authenticatedUserDidGet, emits(user));
+      });
+    });
+
+    describe("its fetchSurveysFromCached is called", () {
+      context(
+          "when surveyRepository's fetchSurveysFromCached to return success",
+          () {
+        final surveys = List<SurveyInfo>.empty();
+        beforeEach(() {
+          when(surveyRepository.fetchSurveysFromCached())
+              .thenAnswer((realInvocation) => Future.value(surveys));
+          interactor.fetchSurveysFromCached();
+        });
+
+        it("triggers delegate's surveysDidFetchFromCached to emit", () {
+          expect(delegate.surveysDidFetchFromCached, emits(surveys));
+        });
+      });
+
+      context(
+          "when surveyRepository's fetchSurveysFromCached to return failure",
+          () {
+        final exception = Exception();
+        beforeEach(() {
+          when(surveyRepository.fetchSurveysFromCached())
+              .thenAnswer((realInvocation) => Future.error(exception));
+          interactor.fetchSurveysFromCached();
+        });
+
+        it("triggers delegate's surveysDidFailToFetchFromCached to emit", () {
+          expect(delegate.surveysDidFailToFetchFromCached, emits(exception));
+        });
+      });
+    });
+
+    describe("its fetchSurveysFromRemote is called", () {
+      context(
+          "when surveyRepository's fetchSurveysFromRemote to return success",
+          () {
+        context(
+            "when surveyRepository's surveysDidFailToFetchFromCached to return outdated data",
```

Comment:
last grammar issue:
`when something happens` -> `when surveyRepository's surveysDidFailToFetchFromCached returns outdated data`.

Same with the other place where you prepare the context

Reply:
@sleepylee updated 😁